provider/equinix/environ_network.go

```diff
@@ -0,0 +1,250 @@
+		}
+
+		if ipblock.Metro == nil || ipblock.Metro.Code != e.cloud.Region {
+			continue
+		}
+
+		subnet := network.SubnetInfo{
+			ProviderId:        network.Id(subnetID),
+			ProviderNetworkId: network.Id(ipblock.ID),
+			CIDR:              cidr, VLANTag: 0,
+			AvailabilityZones: []string{ipblock.Metro.Code},
+		}
+		projectSubnets = append(projectSubnets, subnet)
+	}
+
+	if inst == instance.UnknownId {
+		return projectSubnets, nil
+	}
+
+	device, _, err := e.equinixClient.Devices.Get(string(inst), nil)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var instanceSubnets []network.SubnetInfo
+nextSubnet: //API client limitation since we can't get the actual blocks for individual instance we have to do this
+	for _, psub := range projectSubnets {
+		_, ipnet, err := net.ParseCIDR(psub.CIDR)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+
+		for _, n := range device.Network {
+			if ipnet.Contains(net.ParseIP(n.Address)) {
+				instanceSubnets = append(instanceSubnets, psub)
+				continue nextSubnet
+			}
+		}
+	}
+
+	return instanceSubnets, nil
+}
+
+func makeSubnetIDForNetwork(networkName, address string, mask int) (string, string, error) {
+	_, netCIDR, err := net.ParseCIDR(fmt.Sprintf("%s/%d", address, mask))
+	if err != nil {
+		return "", "", errors.Annotatef(err, "calculating CIDR for network %q", networkName)
+	}
+
+	cidr := netCIDR.String()
+	subnetID := fmt.Sprintf("subnet-%s", networkName)
+	return subnetID, cidr, nil
+}
+
+// SupportsSpaces returns whether the current environment supports
+// spaces. The returned error satisfies errors.IsNotSupported(),
+// unless a general API failure occurs.
+func (e *environ) SupportsSpaces(context.ProviderCallContext) (bool, error) {
+	return true, nil
+}
+
+// NetworkInterfaces returns a slice with the network interfaces that
+// correspond to the given instance IDs. If no instances where found, but there
+// was no other error, it will return ErrNoInstances. If some but not all of
+// the instances were found, the returned slice will have some nil slots, and
+// an ErrPartialInstances error will be returned.
+func (e *environ) NetworkInterfaces(ctx context.ProviderCallContext, ids []instance.Id) ([]network.InterfaceInfos, error) {
+	if len(ids) == 0 {
+		return nil, environs.ErrNoInstances
+	}
+	infos := make([]network.InterfaceInfos, len(ids))
+	for i, id := range ids {
+		device, _, err := e.equinixClient.Devices.Get(string(id), nil)
+		if err != nil {
+			return nil, err
+		}
+
+		dev := &equinixDevice{e, device}
+
+		subnets, err := e.Subnets(ctx, id, nil)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+
+		if len(subnets) == 0 {
+			return nil, errors.Trace(fmt.Errorf("Instance does not have subnets"))
+		}
+
+		deviceAddresses, err := dev.Addresses(ctx)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+
+		for portIdx, port := range device.NetworkPorts {
+			current := network.InterfaceInfo{
+				DeviceIndex:       portIdx,
```

Comment:
```
	// DeviceIndex specifies the order in which the network interface
	// appears on the host. The primary interface has an index of 0.
```

I wonder if we should force `bond0` (or `eth0` if not bonded) into position 0.

Reply:
I believe there is no particular need to force a particular position index for this provider.